### PR TITLE
Migrate to static icons colours

### PR DESCRIPTION
**Pull request by skjnldsv**
Implement a legacy compatibility layer for the svg icons colours
- [x] First step before https://github.com/nextcloud/server/pull/32261
- [x] Split to easier reviews
- [x] Remove SvgController
- [x] Remove IconsCacher
- [x] Implement auto-switch based on theme and/or preferred color scheme

Refs: https://github.com/nextcloud/server/issues/32072 https://github.com/nextcloud/server/pull/32261

### CODE REVIEW

**Comment by CarlSchwan** [comment]
> @CarlSchwan that should drastically remove some load as the svgController was getting pinged a lot for each page load! rocket

I'm not sure how the SvgController works. It seems that before and after this PR there was zero network request against it even after a browser full refresh.

Still, I can see significantly fewer DB queries when loading the cache and the scsscacher is cold, from 375 queries to 210 queries.

**Comment by skjnldsv** [comment]
> Edit: I figured out how the iconcacher works, it basically inject the icons directly inside the CSS by using file_get_contents and then encoding it as base64. So that's why I didn't saw it in my apache logs.

Ah right, we also did that, not always serving those via the svg Controller :thinking: 
I think the svg controller is only used on specific cases, like when you enable the custom dark theme or change some theming values?
We moved the implementation a it, i' don't recall all the usages :see_no_evil:

**Review by CarlSchwan — APPROVED** [review]

**Comment by skjnldsv** [comment]
Yeah, enable dark theme and see the madness :stuck_out_tongue_closed_eyes: 
image

**Comment by skjnldsv** [comment]
Tested a bit, one svg request is 10 DB queries.
Loading files with minimum content is at least 17 svg requests, so it's 170 DB requests saved (most likely much more) :rocket:

File: core/css/functions.scss [hunk 1]
```diff
@@ -36,22 +36,34 @@
...
-		@return '#{$webroot}/svg/core/#{$dir}/#{$icon}?color=#{$color}&v=#{$version}';
-	} @else {
-		@return '#{$webroot}/svg/#{$dir}/#{$icon}?color=#{$color}&v=#{$version}';
+@function match-color-string($color) {
+	@if $color == #000 {
+		@return "dark";
+	}
+	@if $color == #000 {
+		@return 'dark';
+	}
```
Comment (nickvergessen) [inline core/css/functions.scss:47]: 👀👀

File: core/css/functions.scss [hunk 2]
```diff
@@ -26,7 +26,7 @@
  * @param string $color The color
  * @return string The color without #
  */
-@function remove-hash-from-color($color) {
+ @function remove-hash-from-color($color) {
```
Comment (nickvergessen) [inline core/css/functions.scss:29]:
```suggestion
@function remove-hash-from-color($color) {
```

**Review by nickvergessen — APPROVED** [review]
YOLO

**Comment by marekful** [comment]
Hey @skjnldsv 

> I'm not sure how the SvgController works.

The [Svg color api](https://docs.nextcloud.com/server/latest/developer_manual/html_css_design/icons.html) was a way to fetch internal icons (like those in the notifications list) and installed apps' icons in SVG format and optionally in a specified colour. I'm not sure if this API was ever used internally by Nextcloud server or other components (e.g. in relation to themes?). As you can see it's still mentioned in the latest developer guide.

Because this API was public prior to being removed and might have had 3rd party integrations relying on it, it would be nice to offer an alternative if there isn't one already. 

Can you suggest a way to fetch said icons in SVG format (with or without specific colour) or know someone who could?

A few example request URIs of old API:

```
/svg/notifications/notifications-dark?color=d8dee9
/svg/passwords/app-dark?color=d8dee9
/svg/core/places/calendar?color=d8dee9
```

**Pull request merged**